### DISCUSSION TITLE
Octopus Power-up - free 2 hour periods of electricity (in limited areas with lots of renewables)

### DISCUSSION THREAD

**Discussion by greedyboyuk** [BottlecapDave/HomeAssistant-OctopusEnergy#397] [Ideas]
I'm not sure how much media attention there has been for Octopus Power-up because it is a trial and limited geographically at the moment. Octopus trialling with UK Power  providing free electricity for 2 hours where demand is low and renewables producing a lot in a local area. We have a 14:00-16:00 "power up" today.

Is it possible to add the option to factor in a manually added datetime slot for power-ups to be reflected in the data & override the agile data for the time period stated. The zero cost period to then be factored into any calculations. Granted it might be of limited benefit & use, due to be a small audience at the moment plus I also can see the point that it's easy to change automations/scripts/templates to factor this in anyway. After all I now have Octopus Power-up entities that I use for this populated manually by me.

Anyway, food for thought based on a degree of direction of travel. Plus it may be the flipside of the coin when cycling between high demand usage reduction strats such as switch off and save & then cycling into price plunge/power-ups as we saw last Winter.

**Comment by BottlecapDave** [comment]
Hi there. There is already an open issue around [this feature](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues/375). Unfortunately, OE don't currently provide any APIs for this service so it will not be integrated for the time being. I try and limit overrides as I want the integration to be as automated as possible. If one was provided, there is complications when it comes to the frequency of the changes and previous consumption cost calculations.

I'll keep an eye on things, and once an API is available it'll be on my list.

**Comment by PianSom** [comment]
Hello - has there been any change to the API to give access to Power-up info? It’d be great to have analogous sensors to the Saving Session ones.

**Reply by BottlecapDave** [reply]
No change that I can see. My guess is it will be folded into the octoplus programme when/if available. I have been seeing saving sessions with no awarded points which _might_ be them, but then again they might just have been test sessions.

**Reply by PianSom** [reply]
Pity! 

FWIW the last Power-up I had (they don't appear to be universal among eligible participants) was on Sat 18th 11am-2pm. I don't seem to have recorded a Saving Session for then. So there's some evidence to support your theory that they may have been test sessions.

(I'm just in the middle of debugging yaml/automations which will pick up a joined Saving Session and then automate a battery discharge and optionally a battery pre-charge. Hence my odd interest in tracking them!)
